### DISCUSSION TITLE
Using Loopback

### DISCUSSION THREAD

**Discussion by ElliottBridge** [rern/rAudio#846] [Bug Reports]
Hi, I'm trying to use loopback to send the audio from MPD to another application. Loopback is working and everything, it appears when typing play -l. But when trying to add it under the User Configuration tab. I get the error: user configurations: MPD failed with the added lines, Restored to previous configurations.
I add the line:

device "hw:Loopback,1"

to the second box. I'm only used to adding things directly into the text file, so perhaps I missed out on some lines I was meant to add? Although, I'm thinking it may be a bug. Hence the category choice.
Thank you

**Comment by ElliottBridge** [comment]
It seems like MPD is just refusing to "see" the loopback device. I have no idea why, and it's really frustrating me. :(

**Comment by kan-ibal** [comment]
Use numerical mode as
hw:1,0 
where 1 is Loopback card address.
aplay --list-devices

**Reply by ElliottDyson** [reply]
> Use numerical mode as
> hw:1,0 
> where 1 is Loopback card address.
> aplay --list-devices

(Still the same Elliott, just my business account), I'll give that a good go later and see how it goes. Would you just put that in the custom configuration section on the webserver? Or directly add it to a config file?

**Comment by kan-ibal** [comment]
I use it directly from playlist via
mpc add alsa://hw:1,1
and keep it saved in playlist. Then Loopback device is to be seen as input device in another application.
Now I have started to wonder should I understand correctly what you are trying to achieve :). Maybe you need fifo or pipe output plugin?
https://mpd.readthedocs.io/en/stable/plugins.html#output-plugins

**Reply by ElliottDyson** [reply]
I could give FIFO a go if this doesn't work. Never used it before though, also I'd need to check if camillaDSP supports it as an input type

**Comment by Slaughthammer** [comment]
Had the same problem, solution is documented here: https://www.runeaudio.com/forum/alsa-loopback-device-for-dsp-action-t7329.html

I already asked [here](https://github.com/rern/rAudio-1/discussions/753) about that.

**Comment by ElliottDyson** [comment]
> Had the same problem, solution is documented here: https://www.runeaudio.com/forum/alsa-loopback-device-for-dsp-action-t7329.html
> 
> I already asked [here](https://github.com/rern/rAudio-1/discussions/753) about that.

Thank you so much!
